XiaomiSmartHome/XiaomiSmartHome/Program.cs

```diff
@@ -0,0 +1,322 @@
+                        if (data.Model == "gateway")
+                        { 
+                            //// We save the token
+                            gateway.Token = data.Token;
+
+                            //// We deserialize data part
+                            GatewayHeartbeat gateway_data = JsonConvert.DeserializeObject<GatewayHeartbeat>(data.Data);
+
+                            //// If we didn't set gateway's sid
+                            if (gateway.Sid == null)
+                            {
+
+                                //// If data ip equal to gateway ip
+                                if (gateway_data.IP == PackageHost.GetSettingValue<string>("GatewayIP"))
+                                {
+                                    gateway.Sid = data.Sid;
+                                }
+                            }
+
+                            //// We push the gateway SO
+                            PackageHost.PushStateObject<Gateway>("Gateway", gateway);
+
+                        }
+                    }
+
+                    //// If command equal report
+                    else if (data.Cmd == "report")
+                    {
+                        //// We print response if report logs is true
+                        if (PackageHost.GetSettingValue<bool>("ReportLog"))
+                        {
+                            PackageHost.WriteInfo("{0}", response);
+                        }
+
+                        //// If model equal gateway
+                        if ((data.Model == "gateway") && (data.Sid == gateway.Sid))
+                        {
+                            //// We deserialize report data as gateway class
+                            GatewayReport datagate = JsonConvert.DeserializeObject<GatewayReport>(data.Data);
+                            gateway.Report = datagate;
+
+                            //// We push the gateway SO
+                            PackageHost.PushStateObject<Gateway>("Gateway", gateway);
+                        }
+                        else
+                        {
+                            //// If equipement sid is in the equipements list
+                            if (equipements.Contains(data.Sid))
+                            {
+                                SaveEquipement(data.Sid);
+                            }
+                        }
+                    }
+                    Thread.Yield();
+                }
+            }
+            catch(Exception ex)
+            {
+                PackageHost.WriteError("Error Connexion : {0}", ex);
+                return;
+            }
+        }
+
+        /// <summary>
+        /// Get equipements list from the gateway
+        /// </summary>
+        public void GetEquipements()
+        {
+                //// Get equipements list
+                EquipementsList list = FindEquipementsList();
```

Comment:
si tu appelles des méthodes d'instance de ta classe, c'est bien de mettre des "this." devant.
`this.FindEquipementsList();` comme ca en relecture c'est plus clair !